src/transport/TcpTransport.cpp

```diff
@@ -65,6 +65,41 @@ tcpConnectStatus TcpTransport::connect(const string &strServerURL,
   memset(&sin, 0, sizeof(sin));
   sin.sin_family = AF_INET;
   sin.sin_addr.s_addr = inet_addr(hostName.c_str());
+ 
+  if (sin.sin_addr.s_addr == INADDR_NONE) {
+	  struct evutil_addrinfo hints;
+	  struct evutil_addrinfo *answer = NULL;
+	  /* Build the hints to tell getaddrinfo how to act. */
+	  memset(&hints, 0, sizeof(hints));
+	  hints.ai_family = AF_UNSPEC; /* v4 or v6 is fine. */			   
+	  /* Look up the hostname. */
+	  int err = evutil_getaddrinfo(hostName.c_str(), NULL, &hints, &answer);
+	  if (err != 0) {
+		  string info = "Error while resolving " + hostName + ":" + evutil_gai_strerror(err);
+		  THROW_MQEXCEPTION(MQClientException, info, -1);
+	  }
+
+	  struct evutil_addrinfo *ai;
+	  for (ai = answer; ai; ai = ai->ai_next){
+		  char buf[128];
+		  const char *s = NULL;
+		  if (ai->ai_family == AF_INET) {
+			  struct sockaddr_in *sin = (struct sockaddr_in*)ai->ai_addr;
+			  s = evutil_inet_ntop(AF_INET, &sin->sin_addr, buf, 128);
+		  }
+		  else if (ai->ai_family == AF_INET6){
+			  struct sockaddr_in6 *sin6 = (struct sockaddr_in6*)ai->ai_addr;
+			  s = evutil_inet_ntop(AF_INET6, &sin6->sin6_addr, buf, 128);
+		  }
+		  if (s){
+			  sin.sin_addr.s_addr = inet_addr(s);
+			  if (sin.sin_addr.s_addr != INADDR_NONE) {
+				  break;
+			  }
+		  }
+	  }
+  }
+ 
```

Comment:
rebase two commits to only one.

Reply:
format this file with clang-format.

Reply:
There is no need to rebase commits since we use `suqash and merge` button to merge PRs.